### PR TITLE
TST/BUG: Fix and test read_parquet with Cloud filesystem (using S3 / moto)

### PR DESCRIPTION
**Pull request by jorisvandenbossche**

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
This already adds the test infrastructure and a basic roundtrip test, which reproduces the error from https://github.com/geopandas/dask-geopandas/pull/97#issuecomment-901813581 (and then still need to actually fix the bug)

File: dask_geopandas/io/parquet.py [hunk 1]
```diff
@@ -57,9 +58,12 @@ class GeoArrowEngine(ArrowEngine):
     @classmethod
     def read_metadata(cls, *args, **kwargs):
         meta, stats, parts, index = super().read_metadata(*args, **kwargs)
+        fs = args[0]
```
[the remark this answers is not in the document]
Comment (TomAugspurger) [reply dask_geopandas/io/parquet.py:61]: +1 for matching the Dask signature.

**Comment by TomAugspurger** [comment]
I can confirm that the issue I saw in https://github.com/geopandas/dask-geopandas/pull/28#issuecomment-917464374 is fixed with with this PR and this diff

```diff
diff --git a/dask_geopandas/io/parquet.py b/dask_geopandas/io/parquet.py
index fc08f60..786ae9b 100644
--- a/dask_geopandas/io/parquet.py
+++ b/dask_geopandas/io/parquet.py
@@ -38,8 +38,11 @@ def _get_partition_bounds(part, fs):
                 pq_metadata = ParquetFile(f).metadata
     if pq_metadata is None:
         return None
+    pa_metadata = pq_metadata.metadata
+    if pa_metadata is None:
+        return None
 
-    metadata_str = pq_metadata.metadata.get(b"geo", None)
+    metadata_str = pa_metadata.get(b"geo", None)
     if metadata_str is None:
         return None
```

I was reading a parquet file produced by a system that apparently doesn't write metadata (not sure if `pa_metadata` is the right name for that).

File: dask_geopandas/io/parquet.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jorisvandenbossche) [reply dask_geopandas/io/parquet.py]: Yes, I only matched the positional args of the signature now, and passing through the keyword ones as `kwargs`

**Comment by jorisvandenbossche** [comment]
Thanks Tom for checking!

> I was reading a parquet file produced by a system that apparently doesn't write metadata (not sure if pa_metadata is the right name for that).

How can it then be read as a GeoDataFrame? (as I think we now require metadata to be present to be able to create a GeoDataFrame with a geometry column name; in geopandas we detect the case without geo metadata, and then raise an error pointing to pandas.read_parquet)

It's also not easy to reproduce, as I don't think I can write a parquet file without metadata with pyarrow (it always adds an "ARROW:schema" key).

**Comment by jorisvandenbossche** [comment]
I noticed though that `geopandas.read_parquet` will actually not raise the proper custom error when there is not pandas metadata present -> fixing that in https://github.com/geopandas/geopandas/pull/2111

**Review by martinfleis — APPROVED** [review]
Looks good to me!

**Comment by TomAugspurger** [comment]
> How can it then be read as a GeoDataFrame?

Good point :) I was testing out too many things at once and didn't realize I was using pandas for that dataset.

**Comment by martinfleis** [comment]
There seems to be an issue with `38-latest` env that struggles to build. I've cancelled one attempt before and this one is stuck as well.

**Comment by jorisvandenbossche** [comment]
Hmm, I tried to create the `38-latest` env locally, and that seems to work fine. Not sure what's happening, or if one of the added packages give conflicts. Will restart once more.

**Comment by jorisvandenbossche** [comment]
Again the same issue .. I will just remove the new dependencies from this specific environment (or install them with pip) so we can move forward here

**Comment by martinfleis** [comment]
All looks fine now. (I'll let you merge it if you wanted something else here).

**Pull request merged**